ecommerce/courses/publishers.py

```
@@ -29,11 +33,23 @@ def get_course_verification_deadline(self, course):
     def serialize_seat_for_commerce_api(self, seat):
         """ Serializes a course seat product to a dict that can be further serialized to JSON. """
         stock_record = seat.stockrecords.first()
+        try:
+            enrollment_code = Product.objects.filter(
+                product_class__name=ENROLLMENT_CODE_PRODUCT_CLASS_NAME,
+                course__id=seat.attr.course_key
+            ).get(
+                attributes__name='seat_type',
+                attribute_values__value_text=seat.attr.certificate_type
+            )
+            bulk_sku = StockRecord.objects.get(product=enrollment_code).partner_sku
+        except (StockRecord.DoesNotExist, Product.DoesNotExist):
```

If you get `StockRecord.DoesNotExist`, you have a serious problem with the product catalogue. At a minimum, the error should be logged if you're catching it. However, I wouldn't catch it. Let it bubble up, forcing someone to investigate.